pkg/ws/report/handlers/vector_update.go

```diff
@@ -29,14 +32,28 @@ func (h *VectorUpdateHandler) Handle(msg common.Message, client interfaces.IRepo
 
 	var vectorUpdateMsg VectorUpdateMessage
 	if err := json.Unmarshal(msg.Payload, &vectorUpdateMsg); err != nil {
-		return fmt.Errorf("failed to unmarshal VectorUpdateMessage payload: %w", err)
+		return customerrors.NewParseError("failed to unmarshal VectorUpdateMessage payload", err)
 	}
 
 	wt, ok := enums.ParseWeaknessFromString(vectorUpdateMsg.VulnerabilityTypeName)
 	if !ok {
-		return fmt.Errorf("weakness type not found: %s", vectorUpdateMsg.VulnerabilityTypeName)
+		return customerrors.NewParseError("weakness type not found", errors.New(vectorUpdateMsg.VulnerabilityTypeName))
 	}
 
 	client.UpdateVector(wt, vectorUpdateMsg.NewValue)
+
+	solved, _ := reportutils.FilterVulnerabilitiesByStatus(client.GetHubReport().GetRoomVulnerabilities(client.GetRoomID()), client.GetVectorStatus())
+	vectorUpdateResponse := dto.VectorUpdateReponse{
+		ExpectedGlobalCVSSScore:            reportutils.GetGlobalCVSSScore(solved),
+		ExpectedGlobalTotalVulnerabilities: reportutils.GetGlobalTotalVulnerabilities(solved),
+	}
+	responseBytes, err := json.Marshal(vectorUpdateResponse)
+	if err != nil {
+		slog.Error("Failed to marshal vector update response", slog.Any("error", err))
+		return customerrors.NewServerSideError("failed to marshal vector update response")
+	}
```

Comment:
The overall code-block logic is good, we want to

1. Get UNSOLVED vulns (think of these as the vulns that remain after changes are applied) 
2. Calculate GlobalCVSSScore over unsolved vulns
3. Calculate TotalVulns over unsolved vulns

So pretty much you're missing:

```suggestion
	_, notSolved := reportutils.FilterVulnerabilitiesByStatus(client.GetHubReport().GetRoomVulnerabilities(client.GetRoomID()), client.GetVectorStatus())
	vectorUpdateResponse := dto.VectorUpdateReponse{
		ExpectedGlobalCVSSScore:            reportutils.GetGlobalCVSSScore(notSolved),
		ExpectedGlobalTotalVulnerabilities: reportutils.GetGlobalTotalVulnerabilities(notSolved),
	}
	responseBytes, err := json.Marshal(vectorUpdateResponse)
	if err != nil {
		slog.Error("Failed to marshal vector update response", slog.Any("error", err))
		return customerrors.NewServerSideError("failed to marshal vector update response")
	}
```